requirements-dev.txt

```diff
@@ -24,6 +24,8 @@ time-machine==2.16.0
 nbconvert==7.16.6
 
 # mypy types
+scipy-stubs==1.16.0.2; python_version >= '3.11'
+scipy-stubs==1.15.3.0; python_version < '3.10'
```

Comment:
I feel this is a dangerous approach
will 1.16.x features be backported to 1.15.x?

Otherwise we'd have diverging type-checking (depending on the python version, one version might be correct while the other isn't) - so i think this has the potential to cause more problems than the benefit it could bring.

Reply:
I'll change this to `scipy-stubs==1.15.3.0` then, so that it matches the current scipy requirement.